spigot/src/main/kotlin/com/mineinabyss/geary/minecraft/dsl/GearyExtension.kt

```diff
@@ -24,147 +26,181 @@ import org.reflections.util.FilterBuilder
+        private fun getReflections(): Reflections? {
+            // cache the object we get because it takes considerable amount of time to get
+            val cacheKey = CacheKey(this@GearyExtension.plugin, path, excluded)
+            reflectionsCache[cacheKey]?.let { return it }
+            val classLoader = this@GearyExtension.plugin::class.java.classLoader
+            val reflections = Reflections(
+                ConfigurationBuilder()
+                    .addClassLoaders(classLoader)
+                    .addUrls(ClasspathHelper.forClassLoader(classLoader))
+                    .addScanners(SubTypesScanner())
+                    .filterInputsBy(FilterBuilder().apply {
+                        if (path != null) includePackage(path)
+                        excluded.forEach { excludePackage(it) }
+                    })
             )
+
+            reflectionsCache[cacheKey] = reflections
+
+            // Check if the store is empty. Since we only use a single SubTypesScanner, if this is empty
+            // then the path passed in returned 0 matches.
+            if (reflections.store.keySet().isEmpty()) {
+                logWarn("Autoscanner failed to find classes for ${this@GearyExtension.plugin.name}${if (path == null) "" else " in package ${path}}"}.")
+                return null
+            }
+            return reflections
         }
-    }
 
-    /**
-     * Helper function to register sreializers via scanning for geary classes.
-     */
-    @Suppress("UNCHECKED_CAST")
-    @InternalSerializationApi
-    private inline fun <reified T : Any> autoscan(
-        plugin: Plugin,
-        path: String? = null,
-        crossinline registrationFn: (kClass: KClass<T>, serializer: KSerializer<T>, moduleBuilder: PolymorphicModuleBuilder<T>) -> Unit
-    ) {
-        serializers {
-            polymorphic(T::class) {
-                val reflections = Reflections(
-                    ConfigurationBuilder()
-                        .addClassLoaders(plugin::class.java.classLoader)
-                        .addUrls(ClasspathHelper.forClassLoader(plugin::class.java.classLoader))
-                        .addScanners(SubTypesScanner())
-                        .apply {
-                            path?.let {
-                                filterInputsBy(FilterBuilder().includePackage(path))
-                            }
-                        }
-                )
 
-                // Check if the store is empty. Since we only use a single SubTypesScanner, if this is empty
-                // then the path passed in returned 0 matches.
-                if (reflections.store.keySet().isNotEmpty()) {
-                    val components = reflections.getSubTypesOf(T::class.java).toSet()
-                    components.map { it.kotlin }
+        /** Helper function to register serializers via scanning for geary classes. */
+        @InternalSerializationApi
+        public fun <T : Any> registerSerializers(
```

Comment:
Does this need to be public?

Reply:
Same goes for this, otherwise we cant access the function through the inline autoscan function.